### PR TITLE
scripts: Fix codegen to support VK_KHR_format_feature_flags2

### PR DESCRIPTION
**Pull request by djdeath**
If the implementation reports that it supports this feature, it should
fill out VkFormatProperties3KHR properly.

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
@charles-lunarg @mikes-lunarg not sure who the appropriate person is to review these; your names came up as suggestions :)

https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3594 is currently blocked on this.

**Comment by KarenGhavam-lunarG** [comment]
> @charles-lunarg @mikes-lunarg not sure who the appropriate person is to review these; your names came up as suggestions :)

For sure, @charles-lunarg should review

File: icd/generated/mock_icd.cpp [hunk 1]
```diff
@@ -2707,6 +2707,12 @@ static VKAPI_ATTR void VKAPI_CALL GetPhysicalDeviceFormatProperties2KHR(
     VkFormatProperties2*                        pFormatProperties)
 {
     GetPhysicalDeviceFormatProperties(physicalDevice, format, &pFormatProperties->formatProperties);
+    VkFormatProperties3KHR *props_3 = lvl_find_mod_in_chain<VkFormatProperties3KHR>(pFormatProperties->pNext);
+    if (props_3) {
+        props_3->linearTilingFeatures = pFormatProperties->formatProperties.linearTilingFeatures;
+        props_3->optimalTilingFeatures = pFormatProperties->formatProperties.optimalTilingFeatures;
+        props_3->bufferFeatures = pFormatProperties->formatProperties.bufferFeatures;
```
Comment (charles-lunarg) [inline icd/generated/mock_icd.cpp:2714]: This does mean that the mock_icd couldn't set bits not found in the 32 bit format properties flags, but that can be left as a todo for later enhancements to mock_icd.

Comment (charles-lunarg) [reply icd/generated/mock_icd.cpp:2714]: Yeah, I think the bottom 32 bits should be the same and in this case is a reasonable assumption to make to get tests working again.

**Comment by charles-lunarg** [comment]
@djdeath Ah, yes, The CLA does need to be signed before we can merge the code. That's the only thing holding up this PR now.

**Comment by djdeath** [comment]
> @djdeath Ah, yes, The CLA does need to be signed before we can merge the code. That's the only thing holding up this PR now.

Sure I can sign it, but just note this is a contribution from my employer : Intel Corporation :)

**Comment by djdeath** [comment]
> > @djdeath Ah, yes, The CLA does need to be signed before we can merge the code. That's the only thing holding up this PR now.
> 
> Sure I can sign it, but just note this is a contribution from my employer : Intel Corporation :)

Hmm.. CLA signed, not sure why it's not showing up.

**Comment by charles-lunarg** [comment]
Argh, thats very obnoxious. Considering its an Intel pr, I can definitely accept it.

Except, now its telling me there are merge conflicts. Probably from the most recent header update. If that is addressed I can merge.

**Comment by djdeath** [comment]
> Argh, thats very obnoxious. Considering its an Intel pr, I can definitely accept it.
> 
> Except, now its telling me there are merge conflicts. Probably from the most recent header update. If that is addressed I can merge.

Do you have local changes not yet on the master branch? Because this PR is already on top of master.

**Comment by charles-lunarg** [comment]
Okay I may have misread the message. It did *not* have a green button, so I must of read that as 'there are issues'. Re-reading it turns out to say 'there are no issues'.

**Pull request merged**